src/pages/MyTokens/Filters/utils/attributes.ts

```diff
@@ -0,0 +1,138 @@
+import { Attribute, AttributeCount, OfferTokenAttribute } from 'api/restApi/offers/types';
+import { NFTToken } from 'api/uniqueSdk/types';
+import { AttributeType, DecodedAttributes } from '@unique-nft/sdk/tokens';
+import { TFilterAttributes } from '../types';
+
+export const sortAttributeCounts = (attributeCountA: AttributeCount, attributeCountB: AttributeCount) => attributeCountA.numberOfAttributes > attributeCountB.numberOfAttributes ? 1 : -1;
+
+export const attributes = (attributeA: Attribute, attributeB: Attribute) => attributeA.key.localeCompare(attributeB.key);
+
+export const toTokenAttributes = (offerAttributes: OfferTokenAttribute[]) => {
+  const result: DecodedAttributes = { };
+  offerAttributes.map(({ key, value, type }, index) => {
+    const convertedValue = Array.isArray(value) ? value.map((v) => ({ _: v })) : { _: value };
+    result[index] = {
+      isArray: Array.isArray(value),
+      isEnum: type === 'Enum',
+      value: convertedValue,
+      name: { _: key },
+      rawValue: convertedValue,
+      type: type as AttributeType
+    };
+    return result;
+  });
+  return result;
+};
+
+export const getAttributesFromTokens = (tokens: NFTToken[]) => {
+  const attributesMap: { [key: string]: any[] } = {};
+  const attributesForFilter: { [key: string]: Array<{ key: string, count: number }>} = {};
+
+  tokens.forEach(({ attributes }) => {
+    if (attributes) {
+      for (const i in attributes) {
+        const name = attributes[i].name;
+        const value = attributes[i].value;
+        if (attributesMap[name._.toLowerCase()]) {
+          if (Array.isArray(value)) {
+            attributesMap[name._.toLowerCase()] = [...attributesMap[name._.toLowerCase()], ...value.map((attr) => attr._)];
+          } else if (attributes[i].isEnum) {
+            attributesMap[name._.toLowerCase()].push(value._ || value);
+          }
+        } else {
+          if (Array.isArray(value)) {
+            attributesMap[name._.toLowerCase()] = value.map((attr) => attr._);
+          } else if (attributes[i].isEnum) {
+            attributesMap[name._.toLowerCase()] = [value._ || value];
+          }
+        }
+      }
+      for (const attrName in attributesMap) {
+        const counterMap: any = {};
+        attributesMap[attrName].map((attrValue) => {
+          if (counterMap[attrValue]) counterMap[attrValue]++;
+          else counterMap[attrValue] = 1;
+          return attrValue;
+        });
+        attributesForFilter[attrName] = [];
+        for (const attribute of Object.keys(counterMap)) {
+          attributesForFilter[attrName].push({ key: attribute, count: counterMap[attribute] });
+        }
+      }
+    }
+  });
+
+  return attributesForFilter;
+};
+
+export function countTokenAttributes(attributes: DecodedAttributes | undefined): number {
+  let count = 0;
+  if (attributes) {
+    for (const i in attributes) {
```

Comment:
No, because attributes is an object
```
declare type DecodedAttributes = {
    [K: number]: {
        name: LocalizedStringWithDefault;
        value: LocalizedStringOrBoxedNumberWithDefault | Array<LocalizedStringOrBoxedNumberWithDefault>;
        type: AttributeType;
        isArray: boolean;
        rawValue: EncodedTokenAttributeValue;
        isEnum: boolean;
    };
};
```